### DISCUSSION TITLE
energy savings during nighttime

### DISCUSSION THREAD

**Discussion by arnjan** [tbnobody/OpenDTU#265] [General]
Hi, the esp32 offers energy saving. Is it possible to implement energy saving during the night and restart in the morning?
Just like https://www.heise.de/blog/ESP32-to-go-4452689.html

**Comment by tbnobody** [comment]
Unfortunatly thats not possible because even the web UI would be not available when the device is in deep sleep.

**Reply by arnjan** [reply]
that´s true, but the inverters do not offer any information when they are offline.

**Reply by kpwg** [reply]
inverters are also used for nighttime feeding from battery.

**Reply by dj9717** [reply]
The feature can be switched on and off manually. That's what ahoy does for the WR connection, for example.
Gunnar

**Comment by tbnobody** [comment]
Just a calculation example what will happen if the polling during the night will not occour. 
NRF24 has a peak TX power of 30mA @ 3,3V
This equals 0,099W. 
0,099W times 11 hours / day where you can turn the TX off (in average over the year), times 365 days / year, equals 400Wh per year. 
Based on the average price per kWh of 32 cent / kWh in Germany this would result in a saving of ~13cent / year. 
This would only be the case if the NRF24 would send all the time within the 11 hours. Lets assume it sends only 20% (which is also a lot) of the time then you are saving 2,6cent / year... Ok round it up to 5 cent because the NRF24 module is is consuming power in RX mode as well....

**Reply by dj9717** [reply]
Not just the polling, that was just a reference to an example. It's about deep sleep mode. The total consumption would then be around 2.5 µA.
Please don't just look at one user. The sum of a few mA is already a few A for 1000 users.

**Reply by tbnobody** [reply]
Then just look at the user experience... You see the software the first time... You install it in the evening because you are working during the day.... You see a power saving option and think that it's a good idea to enable it... And then you are locked out... nothing happens any more. Maybe you realize that you can access the device in the next morning when the sun is shining again... or maybe you just reset the device. Erase the flash and upload it again and don't turn on that feature anymore...

**Reply by dj9717** [reply]
Of course there are all sorts of things. However, I don't see any argument for exclusion here.

But one thing is still very important. Thanks for the really useful OpenDTU.

**Reply by sapiency** [reply]
hmm, I've noticed in the console that the RF module retries if there is no ACK .. If I look at the console at night I see:

`TX SystemConfigPara Channel: 3 --> xxxxxx
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 23 --> xxxxxx
RX Period End
All missing
Nothing received, resend count exeeded
Fetch inverter: xxxxxxxxxxxx
Request SystemConfigPara
TX RealTimeRunData Channel: 40 --> xxxxxxxxx
RX Period End
All missing
Nothing received, resend whole request`

maybe it makes sense to take account in this ...
BR

**Reply by huckes** [reply]
Yes, the NRF fires about 40 times each nightly try, so a mass more than during daily operation. Seems a bit senceless but doesnt really hurt technically.
Alternatively you can put the power supply of the DTU in a smart outlet to shut down the DTU between sunset and sunrise.

**Comment by gcPhil** [comment]
There are probably people who would prefer an optional night shutdown due to the reduced radio usage. On the other hand, they could use Ahoy DTU, which seems to have the requested function (with calculation of sunrise and sunset times based on position data).

**Reply by dj9717** [reply]
No not for that "reduced radio usage".  But for the ESP32 "Deep Sleep". That really saves and unfortunately that can't be done ahoy either.

**Reply by olivierschmid** [reply]
Alternatively we could incrase the polling time interval to for example 1 minute when going to "sleep mode" (as long there is no communication with inverter). When communication comes back in the morning set the polling interval back to "normal". Of course that would be not the big power safer but reduces at least wlan traffic :-)

**Comment by joschua234** [comment]
Here someone (@LarsRaw) has built a great solution for the deep sleep at night. I have this version installed and I highly recommend it.
https://github.com/tbnobody/OpenDTU/issues/397#issuecomment-1445438135
